### ISSUE TITLE
scalafix.v1.Symbol#normalized does not handle quoted identifier containing dots properly

### ISSUE THREAD

**Issue by liancheng** [scalacenter/scalafix#1097]
The following patch contains a test case that reproduces this issue:

```diff
diff --git a/scalafix-tests/shared/src/main/scala/test/SymbolTest.scala b/scalafix-tests/shared/src/main/scala/test/SymbolTest.scala
new file mode 100644
index 00000000..791df13b
--- /dev/null
+++ b/scalafix-tests/shared/src/main/scala/test/SymbolTest.scala
@@ -0,0 +1,9 @@
+package test
+
+import a.`b.c`.d
+
+object a {
+  object `b.c` {
+    object d
+  }
+}
diff --git a/scalafix-tests/unit/src/test/scala/scalafix/tests/v1/SymbolSuite.scala b/scalafix-tests/unit/src/test/scala/scalafix/tests/v1/SymbolSuite.scala
new file mode 100644
index 00000000..9a7491ef
--- /dev/null
+++ b/scalafix-tests/unit/src/test/scala/scalafix/tests/v1/SymbolSuite.scala
@@ -0,0 +1,19 @@
+package scalafix.tests.v1
+
+import org.scalatest.FunSuite
+import scala.meta.Import
+import scala.meta.Importer
+import scalafix.tests.core.BaseSemanticSuite
+import scalafix.v1._
+
+class SymbolSuite extends FunSuite {
+  implicit val doc = BaseSemanticSuite.loadDoc("SymbolTest.scala")
+
+  test("owner") {
+    val Seq(Import(Seq(Importer(ref, _)))) = doc.tree.collect {
+      case t: Import => t
+    }
+
+    assert(ref.symbol.normalized.owner.value == "test.a.")
+  }
+}
```

Test failure output:

```
[info] SymbolSuite:
[info] - owner *** FAILED *** (281 milliseconds)
[info]   "test.a.[b.]" did not equal "test.a.[]" (SymbolSuite.scala:17)
```

Seems that `Symbol#normalized` does not preserve the quotes and converts `` test.a.`b.c` ``  into `test.a.b.c`

**Comment by liancheng** [comment]
I believe we need to quote `desc.name` when necessary here:

https://github.com/scalacenter/scalafix/blob/abf6e0963b9c203cd3d96b0d34fa5ac8f01a41bd/scalafix-core/src/main/scala/scalafix/internal/util/SymbolOps.scala#L72

The Scalameta [`Scala.Names.encode()` method][1] can be applied here but unfortunately, it's package private. Maybe we can inline it somewhere.

[1]: https://github.com/scalameta/scalameta/blob/059f035259e180aa6bfee4b70fd5f16f3155ea7e/semanticdb/semanticdb/src/main/scala/scala/meta/internal/semanticdb/Scala.scala#L175-L184

**Comment by olafurpg** [comment]
Thank you for reporting! I think you are correct that the fix is to encode `desc.name`. We can access the package private method by adding a forwarder method here https://github.com/scalacenter/scalafix/blob/abf6e0963b9c203cd3d96b0d34fa5ac8f01a41bd/scalafix-core/src/main/scala/scala/meta/internal/scalafix/ScalafixScalametaHacks.scala#L13 It's not pretty but avoids copy-pasting code. I feel like there are too many `private[meta]` methods for no good reason, the code already lives in the `scala.meta.internal` package so there are no guarantees about binary compatibility anyways

**Comment by liancheng** [comment]
Ah, the forwarder method is a nice trick! Let me have a try.

**Comment by liancheng** [comment]
Fixed by #1100.
